dev.c

```diff
@@ -281,12 +315,24 @@ static void fuse_request_send_nowait_locked(struct fuse_conn *fc,
 		set_bdi_congested(&fc->bdi, BLK_RW_ASYNC);
 	}
 	fc->active_background++;
+    if (fc->active_background >= fc->accumulate) {
+        fc->active_background = 0;
+        wakeup = true;
+    }
 	req->in.h.unique = fuse_get_unique(fc);
 	queue_request(fc, req);
```

Comment:
If the requests miss the simple sequential merge above, then they are added to the tail... This impacts the merging logic above, if the requests are ordered in ascending order of block addresses, then it is possible to do more merges when out of order requests gets submitted/processed.

Reply:
We could maintain a sorted list in pxd driver, but our current code was not even doing good job when the kernel was sending writes in order.